### ISSUE TITLE
False positive for UnusedOperation when used in a function call made on module attribute module

### ISSUE THREAD

**Issue by michallepicki** [rrrene/credo#1269]
Hello! I am on credo `1.7.17`, elixir `1.19.4-otp-28`, erlang `28.3`

We have some modules where we use module attributes to do "dependency injection", for testing etc, e.g. `@http_client Application.compile_env(:some_app, :some_key, SomeHttpClient)`. I noticed we are getting "UnusedOperation" warnings for arguments passed when calling functions from such injected modules, e.g. for `Credo.Check.Warning.UnusedMapOperation` and this module:

```elixir
defmodule SomeMod do
  @other_mod OtherMod

  def some_function do
    @other_mod.other_function(Map.merge(%{}, %{}))

    :ok
  end
end
```

we see a warning, even though the Map function's return value is used.

**Comment by rrrene** [comment]
@michallepicki Thx so much for reporting this! 🙏  Should now be fixed on `master` :+1:
